src/ucp/core/ucp_context.c

```diff
@@ -48,7 +48,7 @@ static const char * ucp_rndv_modes[] = {
 
 uct_memory_type_t ucm_to_uct_mem_type_map[] = {
     [UCM_MEM_TYPE_CUDA]         = UCT_MD_MEM_TYPE_CUDA,
-    [UCM_MEM_TYPE_CUDA_MANAGED] = UCT_MD_MEM_TYPE_HOST
+    [UCM_MEM_TYPE_CUDA_MANAGED] = UCT_MD_MEM_TYPE_CUDA_MANAGED
```

Comment:
do we need this change in this PR? as such this PR not adding any special additions in case of UCT_MD_MEM_TYPE_CUDA_MANAGED.  we can add this when are adding  functionality which differentiating from HOST.

Reply:
@bureddy Agree with you. Had a discussion internally as well and came to a conclusion that we could add a new UCT that supports MD_MEM_TYPE_CUDA_MANAGED and provides put_short (using memcpy) so that mem_type_pack and mem_type_unpack picks that UCT while using am_bcopy. If this sounds reasonable, I'll follow up with changes.

Reply:
hmm I'm no sure if we need to go with new UCT. just saying that currently in this PR change [UCM_MEM_TYPE_CUDA_MANAGED] = UCT_MD_MEM_TYPE_CUDA_MANAGED not adding any new functionality. right?

Reply:
I may be wrong but I think it is still needed. In the function `ucp_memory_type_detect_mds`
in `ucp_context.h:304`, `mem_type` gets assigned to `UCT_MD_MEM_TYPE_CUDA_MANAGED` using the above mapping in the statement `*mem_type_p = ucm_to_uct_mem_type_map[ucm_mem_type];` This is after pointer cache lookup.
This is needed dt.c, dt.inl, ucp_request.inl when the macro `UCP_MEM_IS_CUDA_MANAGED` is called. Can we still assign the right mem_type_p without the above mapping?

Reply:
you need this mapping.  But, this PR is not adding any extra new functionality with new mapping,  it is still same as HOST right?

Reply:
Agreed. This PR isn't adding a new functionality but relying on existing pack/unpack used for HOST. Given that we can't treat UCM_MEM_TYPE_CUDA_MANAGED as host because we don't want zcopy transfer to/from them. Should we remove this piece of code and have a different condition? 

if (mem_type == cuda) { mem_type_pack/unpack} else { memcpy //covers host and managed }

Will this suffice or do you have suggestions on how we should go about handling managed memory?